### DISCUSSION TITLE
Feature Request: Scope Control Enhancement - Allow for Ingestion of In-Scope IPs

### DISCUSSION THREAD

**Discussion by mbb5546** [projectdiscovery/subfinder#1229] [Ideas]
As a security tester I'm a big fan of PD tooling. Subfinder is great and I think it'd be awesome if Subfinder had a scope-control functionality that allowed the user/tester to provide both domains and specific IP addresses/network ranges that have been provided as in-scope. 

A great example of this functionality is BBOT which has the ability to ingest multiple types of targets (URLs, domains, IPs) and while ensuring that scans and results are ultimately respective of the scope provided to the tool.

Scope control would be a really powerful feature and would help bring this tool to the forefront of my testing methodology. I'm sure other's would say the same.

**Comment by GeorginaReeder** [comment]
Moving this to Ideas discussion. Thanks @mbb5546 !

**Comment by dogancanbakir** [comment]
Thanks for the idea. Subfinder sources return only names, so IP/CIDR scoping would require resolving every result, which belongs downstream. Resolve with `dnsx -a -resp` and match IPs against your scope with `mapcidr -mi scope.txt`. Name-based scoping is available via `-m` / `-f`.
